### PR TITLE
Add README file for Helm charts section

### PR DESCRIPTION
**Pull request by MaxBab**
- Add instructions on deploying the sail-operator by using the helm charts defined within the repository.
- Rearrange the following guides to avoid duplication.
  Place them within 'docs/common' directory and reference them from the relevant docs guides.
    - create-and-configure-gateways.md
    - install-bookinfo-app.md
    - install-istioctl-tool.md
    - istio-addons-integrations.md

Fixes https://github.com/istio-ecosystem/sail-operator/issues/314

### CODE REVIEW

File: chart/README.md
Comment (MaxBab) [reply chart/README.md]: Fixed

File: chart/README.md
Comment (MaxBab) [reply chart/README.md]: Modified.

File: chart/README.md
Comment (MaxBab) [reply chart/README.md]: Changed and added.

**Comment by fjglira** [comment]
Adding hold, in case you want to do any other change of someone else want to take a look

File: chart/README.md
Comment (sridhargaddam) [inline chart/README.md:1]: JFYI, we are using the convention Sail Operator in our docs.

```suggestion
# Deploy Sail Operator by using Helm charts
```

File: chart/README.md
Comment (sridhargaddam) [inline chart/README.md:38]:
```suggestion
The variables specified in the command are as follows:
```

File: chart/README.md
Comment (sridhargaddam) [inline chart/README.md:51]:
```suggestion
**Note** - This step could be skipped by using the `--create-namespace` argument in step 2.
```

File: chart/README.md
Comment (sridhargaddam) [inline chart/README.md:185]: Since this content is similar to https://github.com/istio-ecosystem/sail-operator/tree/main/bundle#installing-the-istioctl-tool, it would be good if we avoid the duplication by pointing to the appropriate page (or re-arranging the content to its own page and linking them where necessary).

File: chart/README.md
Comment (MaxBab) [reply chart/README.md]: Fixed

File: chart/README.md
Comment (MaxBab) [reply chart/README.md]: Fixed

File: chart/README.md
Comment (MaxBab) [reply chart/README.md:185]: That's a good point.

I made some re-arragement of the docs to avoid such duplication.

File: chart/README.md
Comment (sridhargaddam) [inline chart/README.md:108]:
```suggestion
* `Istio`.
* If you are using OpenShift, the `IstioCNI` must also be created.

The `Istio` resource deploys and configures the Istio Control Plane, whereas the `IstioCNI` resource (in OpenShift) deploys and configures the Istio CNI plugin. You should create these resources in separate projects.
```

File: chart/README.md
Comment (sridhargaddam) [inline chart/README.md:142]:
```suggestion
**Note** - The version can be specified by modifying the `version` field within `Istio` and `IstioCNI` manifests.
```

**Review by sridhargaddam — COMMENTED** [review]
Thanks @MaxBab for re-arranging the pages and removing the duplication.

File: bundle/README.md
Comment (MaxBab) [reply bundle/README.md]: Modified.

File: bundle/README.md
Comment (MaxBab) [reply bundle/README.md]: Modified.

File: bundle/README.md
Comment (MaxBab) [reply bundle/README.md]: Modified.

File: chart/README.md
Comment (MaxBab) [reply chart/README.md]: Modified.

**Comment by sridhargaddam** [comment]
LGTM. I'll wait for sometime if someone wants to review this PR before merging.

**Review by fjglira — APPROVED** [review]
LGTM

**Comment by fjglira** [comment]
@MaxBab, you can delete the hold label when you want to merge the PR :)

**Pull request merged**
